packages/react-components/react-list-preview/src/components/List/useList.ts

```diff
@@ -9,23 +25,72 @@ import type { ListProps, ListState } from './List.types';
  * before being passed to renderList_unstable.
  *
  * @param props - props from this instance of List
- * @param ref - reference to root HTMLDivElement of List
+ * @param ref - reference to root HTMLElement of List
  */
-export const useList_unstable = (props: ListProps, ref: React.Ref<HTMLDivElement>): ListState => {
+export const useList_unstable = (props: ListProps, ref: React.Ref<HTMLDivElement | HTMLUListElement>): ListState => {
+  const { navigationMode, selectionMode, selectedItems, defaultSelectedItems, as, onSelectionChange } = props;
+  const haveListItemsBeenValidated = React.useRef(false);
+
+  const arrowNavigationAttributes = useArrowNavigationGroup({
+    axis: 'vertical',
+    memorizeCurrent: true,
+  });
+
+  const [selectionState, setSelectionState] = useControllableState({
+    state: selectedItems,
+    defaultState: defaultSelectedItems,
+    initialState: [],
+  });
+
+  const onChange = useEventCallback((e: React.SyntheticEvent, data: OnSelectionChangeData) => {
+    const selectedItemsAsArray = Array.from(data.selectedItems);
+    setSelectionState(selectedItemsAsArray);
+    onSelectionChange?.(e, { event: e, type: 'change', selectedItems: selectedItemsAsArray });
+  });
+
+  const selection = useListSelection({
+    onSelectionChange: onChange,
+    selectionMode: selectionMode || 'multiselect',
+    selectedItems: selectionState,
+    defaultSelectedItems,
+  });
+
+  const listRole = props.role || calculateListRole(navigationMode, !!selectionMode);
+  const listItemRole = calculateListItemRoleForListRole(listRole);
+
+  const { findAllFocusable } = useFocusFinders();
+
+  const validateListItems = useEventCallback((listItemEl: HTMLElement) => {
+    if (!haveListItemsBeenValidated.current) {
+      const itemRole = listItemEl.getAttribute('role') || '';
+      const focusable = findAllFocusable(listItemEl);
+      validateProperRolesAreUsed(listRole, itemRole, !!selectionMode, focusable.length > 0);
+      validateGridCellsArePresent(listRole, listItemEl);
+      haveListItemsBeenValidated.current = true;
+    }
```

Comment:
Why not put this to `useEffect`? It will run once...

Reply:
fixed